### PR TITLE
chore: add checks and formatting

### LINKED ISSUE #54: Add checks and formatting
A proper CI of a Rust library would tipically expect to contain as minimum a `cargo check` and a `cargo clippy` execution.

### PR DESCRIPTION
**Pull request by mondeja**
Closes #54

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
```
Comment (friendlymatthew) [inline .github/workflows/ci.yml:6]: should this be main?

**Review by friendlymatthew — APPROVED** [review]
Lgtm -- just one thing

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mondeja) [reply .github/workflows/ci.yml]: Ups, yes! 🦭

**Comment by friendlymatthew** [comment]
We should also run these CI's with the `--features=debug` command, this way it won't think we're in SSR mode

**Comment by mondeja** [comment]
> We should also run these CI's with the `--features=debug` command, this way it won't think we're in SSR mode

Nope. The feature `debug` can be added separately from `ssr`, so I'm checking apart. The problem is that there is inconsistencies on the code between features. I've resolved most of them, could you handle the rest that you're more confident with the codebase?

Note some changes that I've done like not importing `web_sys` (only instantiating it when needed). Keep in mind that code inside `create_effect` only run on the client, so there is no need to wrap it with `ssr` checks.

File: leptos_hotkeys/src/hotkeys_provider.rs [hunk 2]
```diff
@@ -43,15 +32,16 @@ where
...
     #[cfg(not(feature = "ssr"))]
-    let set_ref_target = Callback::new(move |target: Option<EventTarget>| {
+    let set_ref_target = Callback::new(move |target: Option<web_sys::EventTarget>| {
         active_ref_target.set(target);
     });
 
     #[cfg(not(feature = "ssr"))]
-    let pressed_keys: RwSignal<HashMap<String, KeyboardEvent>> = RwSignal::new(HashMap::new());
+    let pressed_keys: RwSignal<HashMap<String, web_sys::KeyboardEvent>> =
+        RwSignal::new(HashMap::new());
```
Comment (mondeja) [inline leptos_hotkeys/src/hotkeys_provider.rs:44]: The main problem is this. When `ssr`, this code should also run.

**Comment by mondeja** [comment]
I'm going to merge to separate things. Thanks for the review!

File: leptos_hotkeys/src/hotkeys_provider.rs (hunk 2 above)
Comment (friendlymatthew) [reply leptos_hotkeys/src/hotkeys_provider.rs:44]: thanks!!

**Comment by friendlymatthew** [comment]
thank you!

**Pull request merged**